src/main/java/com/github/pires/hazelcast/HazelcastDiscoveryController.java

```diff
@@ -66,23 +80,43 @@ private static String getEnvOrDefault(String var, String def) {
         : val;
   }
 
+  // TODO: Load the CA cert when it is available on all platforms.
+  private static TrustManager[] trustAll = new TrustManager[] {
+    new X509TrustManager() {
+      public void checkServerTrusted(X509Certificate[] certs, String authType) {}
+      public void checkClientTrusted(X509Certificate[] certs, String authType) {}
+      public X509Certificate[] getAcceptedIssuers() { return null; }
+    }
+  };
+  private static HostnameVerifier trustAllHosts = new HostnameVerifier() {
+    public boolean verify(String hostname, SSLSession session) {
+      return true;
+    }
+  };
+
   @Override
   public void run(String... args) {
-    final String hostName = getEnvOrDefault("KUBERNETES_RO_SERVICE_HOST",
-        "localhost");
-    final String hostPort = getEnvOrDefault("KUBERNETES_RO_SERVICE_PORT",
-        "8080");
     String serviceName = getEnvOrDefault("HAZELCAST_SERVICE", "hazelcast");
     String path = "/api/v1beta3/namespaces/default/endpoints/";
-    final String host = "http://" + hostName + ":" + hostPort;
+    final String host = "https://kubernetes.default.cluster.local";
```

Comment:
Last time I checked kubernetes.local was the default domain. Has this changed? 

Also, what happens if the user specifies a different domain?